wolfcrypt/src/wolfmath.c

```diff
@@ -243,12 +243,20 @@ int wc_mp_to_bigint_sz(mp_int* src, WC_BIGINT* dst, word32 sz)
     if (src == NULL || dst == NULL)
         return BAD_FUNC_ARG;
 
+    /* get size of source */
+    x = mp_unsigned_bin_size(src);
+    if (sz < x)
```

Comment:
What's the purpose of the sz argument if we're going to change it w/o notifying the caller?

Reply:
That's a very good point! It works for Nitrox because internally it uses the right buffer, but there is no reason we couldn't set the internal size and pass the correct size to hardware. I will adjust and retest. Thanks Todd.

Reply:
Correction on my last statement. The size does get set into the `dst->src` via the `wc_bigint_alloc` call. The purpose of the `sz` argument is to make sure the buffer is at least that large and is leading zero padded. Added comment and improved wc_mp_to_bigint case.